packages/shared/src/libs/helpers/get-relative-date/get-relative-date.helper.ts

```diff
@@ -0,0 +1,21 @@
+import { formatRelative } from "date-fns";
+import { enUS } from "date-fns/locale";
+
+const formatRelativeLocale = {
+	lastWeek: "'Updated last' eeee",
+	other: "'Updated on' P",
+	today: "'Updated today'",
+	yesterday: "'Updated yesterday'",
+};
+
+const locale = {
+	...enUS,
+	formatRelative: (token: string): string =>
+		formatRelativeLocale[token as keyof typeof formatRelativeLocale],
+};
+
+const getRelativeDate = (date: Date, baseDate: Date): string => {
+	return formatRelative(date, baseDate, { locale });
+};
```

Comment:
Haven't found anything better for now (I prefer to change it later if requested). Should I also move `formatRelativeLocale` and `locale` to different files in this folder (create libs/enums/format-relative-locale.map.ts and relative-locale.helper.ts), or even outside this folder (not sure it is reusable, so I prefer to leave it as it is).

Reply:
But if we prefer dates, I can add a little complexity like this `"'Updated' " + daysDifference.toString() + " 'days ago"`

```
const formatRelativeLocale = {
	lastWeek: "'Updated last' eeee",
	other: (date: Date, baseDate: Date): string => {
		const daysDifference = differenceInDays(baseDate, date);

		return "'Updated' " + daysDifference.toString() + " 'days ago";
	},
	today: "'Updated today'",
	yesterday: "'Updated yesterday'",
};

const locale = {
	...enUS,
	formatRelative: (token: string, date: Date, baseDate: Date): string => {
		const formatter =
			formatRelativeLocale[token as keyof typeof formatRelativeLocale];

		return typeof formatter === "function"
			? formatter(date, baseDate)
			: formatter;
	},
};
```

It will show `X days ago`, instead of date

Reply:
Or make it simpler, but with hours

```
import { formatDistance } from "date-fns";

const getRelativeDate = (
	date: Date | number | string,
	baseDate: Date | number | string,
): string => {
	return "Updated " + formatDistance(date, baseDate);
};

export { getRelativeDate };
```
![image](https://github.com/user-attachments/assets/d0d2d605-cdcc-446e-9cbc-ded331849943)

Reply:
Or a little more complicated, but with today and yersterday

```
import { differenceInDays, formatDistance, isSameDay } from "date-fns";

const getRelativeDate = (
	date: Date | number | string,
	baseDate: Date | number | string,
): string => {
	const parsedDate = new Date(date);
	const parsedBaseDate = new Date(baseDate);
	const DIFF_IN_A_DAY = 1;

	if (isSameDay(parsedDate, parsedBaseDate)) {
		return "Updated today";
	}

	if (differenceInDays(parsedBaseDate, parsedDate) === DIFF_IN_A_DAY) {
		return "Updated yesterday";
	}

	return "Updated " + formatDistance(parsedDate, parsedBaseDate) + " ago";
};

export { getRelativeDate };
```
![image](https://github.com/user-attachments/assets/15819562-508d-4eb3-9a0b-11124abf78d2)

Reply:
Also as we are checking for last activity (not last statistics update), we also need to check that there is at least one commit for this project for the date before updating last_activity_date

Reply:
if I understand correctly, even with reset (`startOfDay(new Date())`) we still will get them in answer (but i'll check it once  battery charge)

or you mean to reset, but also add this
```
if (isSameDay(parsedDate, parsedBaseDate)) {
	return "Updated today";
}

if (differenceInDays(parsedBaseDate, parsedDate) === DIFF_IN_A_DAY) {
	return "Updated yesterday";
}
```

Reply:
Let's leave the current approach, looks ok to me